### PR TITLE
Do not use `strtok` in vfxParser

### PR DESCRIPTION
**Pull request by kuhar**
`strtok` relies on global variables and is not thread safe.

This replaces all uses of `strtok` in LLPC with a thread safe helper
function `split`. The semantics are not identical, but the use of `split`  
in the parser code preserves the existing behavior.

Add Vfx unit tests.

Also add macros to guards against future accidental uses of `strtok`.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 4ad7bde60b6b8ea6ac2c088193ff7a52ad3d87db
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 44abc48d199dd28369bf7415eee1751c4bf616d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev 38aa59bcf134d1cb340ddd2459b34b6eda798abe</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 9f7a59928db862a2baded0b45582ce3683d83762</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225376)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32512/56344 (57.7%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23832/56344 (42.3%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32352/56344 (57.4%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23991/56344 (42.6%)</li><li>Warnings: 1/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32688/56344 (58.0%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23656/56344 (42.0%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35669/56344 (63.3%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 20675/56344 (36.7%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by kuhar** [comment]
PTAL @vettoreldaniele @amdrexu @linqun @jiaolu @piotrAMD

**Review by vettoreldaniele — CHANGES_REQUESTED** [review]

File: tool/vfx/vfxParser.cpp [hunk 1]
```diff
@@ -94,6 +101,39 @@ bool isArrayAccess(const char *str);
 // in the string
 char *getWordFromString(char *str, char *wordBuffer);
 
+// =====================================================================================================================
+// Splits the input string by modifying it in place. Returns a vector of (inner) fragment strings. This is a thread-safe
```
[the remark this answers is not in the document]
Comment (kuhar) [reply tool/vfx/vfxParser.cpp]: Thanks, I wrote `split` so that it did what the code expected it to and missed that `strtok` takes a set of delimiters instead of a single one. I changed the code so that treats it as a set of delimiter characters, but I did not pursue 100% compatibility with `strtok` (there are some corner cases around leading/trailing delimiters, empty delimiter sets, etc., that do not matter for Vfx).

I added unit tests like you suggested.

**Comment by kuhar** [comment]
V1: Treat the `delim` string as a set of delimiter characters to match `strtok`. Add unit tests.

**Comment by kuhar** [comment]
retest this please

**Comment by kuhar** [comment]
@JaxLinAMD what's going on with Jenkins?

**Comment by JaxLinAMD** [comment]
internal server is down. I think it need at least half a day to recover.

**Review by vettoreldaniele — APPROVED** [review]

**Review by amdrexu — APPROVED** [review]

**Comment by kuhar** [comment]
retest this please
